sandboxnu/searchneu-v1
# Convert to GraphQl

## What
Converting search to use the GraphQL endpoint instead of Rest. 
See https://github.com/sandboxnu/course-catalog-api/pull/11 as well.

## How
Hit the GraphQL endpoint and transform the results we get back to mirror the shape of the old rest API results.

## Steps To Test
- Make sure `course-catalog-api` is running locally (if the branch `ml-hit-graphql` in `course-catalog-api` hasn't been merged into master yet, switch to that branch)
- In your local search repo, go to the file `codegen.yml` and replace `schema: 'https://searchneu.com/graphql'` with `schema: 'http://localhost:4000'`
- In your local search repo, go to `utils/courseAPIClient.ts` and replace `const ENDPOINT = 'https://searchneu.com/graphql';` with `const ENDPOINT = 'http://localhost:4000'`
